src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioSamplers.java

```diff
@@ -41,19 +40,15 @@ public VarianceSampler(final double varianceMin, final double varianceMax, final
         public Double sample(final RandomGenerator rng, final CopyRatioState state, final CopyRatioData dataCollection) {
             final Function<Double, Double> logConditionalPDF = newVariance -> {
                 double ll = 0.;
-                int numNotOutliers = 0;
+                final double gaussianLogNormalization = 0.5 * Math.log(newVariance);
                 for (int segment = 0; segment < dataCollection.getNumSegments(); segment++) {
-                    final int numTargetsInSegment = dataCollection.getNumTargetsInSegment(segment);
-                    final int targetStartInSegment = dataCollection.getStartTargetInSegment(segment);
-                    for (int target = 0; target < numTargetsInSegment; target++) {
-                        final double coverage = dataCollection.getCoveragesInSegment(segment).get(target);
-                        if (!state.targetOutlierIndicator(targetStartInSegment + target)) {
-                            ll -= normalTerm(coverage, state.segmentMean(segment), newVariance);
-                            numNotOutliers++;
+                    final List<CopyRatioData.IndexedCoverage> indexedCoveragesInSegment = dataCollection.getIndexedCoveragesInSegment(segment);
```

Comment:
i did consider using a stream here (as in the mean sampler below), but i think there's just enough going on that it's clearer to loop over the segments.